File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraScheduler.UI;
using DevExpress.XtraScheduler;
using DevExpress.Services;
using DevExpress.XtraScheduler.Services.Implementation;
using DevExpress.XtraScheduler.Commands;
using DevExpress.XtraScheduler.Native;
using DevExpress.Portable.Input;

namespace HowTo {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }
        void Form1_Load(object sender, EventArgs e) {
            SchedulerKeyboardHandlerService oldService = (SchedulerKeyboardHandlerService)schedulerControl1.Services.GetService(typeof(IKeyboardHandlerService));
            schedulerControl1.Services.RemoveService(typeof(IKeyboardHandlerService));
            MySchedulerKeyboardHandlerService myKeyboardService = new MySchedulerKeyboardHandlerService(oldService);
            schedulerControl1.Services.AddService(typeof(IKeyboardHandlerService), myKeyboardService);
            
        }
    }
    public class MySchedulerKeyboardHandlerService : IKeyboardHandlerService {
        SchedulerKeyboardHandlerService oldService;
        public MySchedulerKeyboardHandlerService(SchedulerKeyboardHandlerService oldService) {
            this.oldService = oldService;
        }
        public SchedulerKeyboardHandlerService OldService { get { return oldService; } }
        #region IKeyboardHandlerService Members       

        public void OnKeyDown(PortableKeyEventArgs e) {
            OldService.OnKeyDown(e);
        }

        public void OnKeyPress(PortableKeyPressEventArgs e) {
            Keys modifier = Form1.ModifierKeys;
            if((modifier & Keys.Alt) == 0 && (modifier & Keys.Control) == 0) {
                SchedulerControl control = (SchedulerControl)oldService.Control.Owner;
                SchedulerCommand command = null;
                if(control.SelectedAppointments.Count <= 0)
                    command = new NewAppointmentCommand(control);
                else if(control.SelectedAppointments.Count == 1)
                    command = new EditAppointmentQueryCommand(control);
                if(command != null) {
                    e.Handled = true;
                    command.Execute();
                }
            }
            else
                OldService.OnKeyPress(e);
        }
        public void OnKeyUp(PortableKeyEventArgs e) {
            OldService.OnKeyUp(e);
        }
        #endregion
    }
}

File: VB/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraScheduler.UI
Imports DevExpress.XtraScheduler
Imports DevExpress.Services
Imports DevExpress.XtraScheduler.Services.Implementation
Imports DevExpress.XtraScheduler.Commands
Imports DevExpress.XtraScheduler.Native
Imports DevExpress.Portable.Input

Namespace HowTo
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
			Dim oldService As SchedulerKeyboardHandlerService = DirectCast(schedulerControl1.Services.GetService(GetType(IKeyboardHandlerService)), SchedulerKeyboardHandlerService)
			schedulerControl1.Services.RemoveService(GetType(IKeyboardHandlerService))
			Dim myKeyboardService As New MySchedulerKeyboardHandlerService(oldService)
			schedulerControl1.Services.AddService(GetType(IKeyboardHandlerService), myKeyboardService)

		End Sub
	End Class
	Public Class MySchedulerKeyboardHandlerService
		Implements IKeyboardHandlerService

'INSTANT VB NOTE: The field oldService was renamed since Visual Basic does not allow fields to have the same name as other class members:
		Private oldService_Renamed As SchedulerKeyboardHandlerService
		Public Sub New(ByVal oldService As SchedulerKeyboardHandlerService)
			Me.oldService_Renamed = oldService
		End Sub
		Public ReadOnly Property OldService() As SchedulerKeyboardHandlerService
			Get
				Return oldService_Renamed
			End Get
		End Property
		#Region "IKeyboardHandlerService Members       "

		Public Sub OnKeyDown(ByVal e As PortableKeyEventArgs) Implements IKeyboardHandlerService.OnKeyDown
			OldService.OnKeyDown(e)
		End Sub

		Public Sub OnKeyPress(ByVal e As PortableKeyPressEventArgs) Implements IKeyboardHandlerService.OnKeyPress
			Dim modifier As Keys = Form1.ModifierKeys
			If (modifier And Keys.Alt) = 0 AndAlso (modifier And Keys.Control) = 0 Then
				Dim control As SchedulerControl = DirectCast(oldService_Renamed.Control.Owner, SchedulerControl)
				Dim command As SchedulerCommand = Nothing
				If control.SelectedAppointments.Count <= 0 Then
					command = New NewAppointmentCommand(control)
				ElseIf control.SelectedAppointments.Count = 1 Then
					command = New EditAppointmentQueryCommand(control)
				End If
				If command IsNot Nothing Then
					e.Handled = True
					command.Execute()
				End If
			Else
				OldService.OnKeyPress(e)
			End If
		End Sub
		Public Sub OnKeyUp(ByVal e As PortableKeyEventArgs) Implements IKeyboardHandlerService.OnKeyUp
			OldService.OnKeyUp(e)
		End Sub
		#End Region
	End Class
End Namespace
